SocialCareCaseViewerApi/V1/UseCase/FormSubmissionsUseCase.cs

```diff
@@ -159,6 +107,76 @@ public CaseSubmissionResponse UpdateAnswers(string submissionId, string stepId,
             return submission.ToDomain().ToResponse();
         }
 
+        private static void UpdateSubmissionState(CaseSubmission submission, UpdateCaseSubmissionRequest request, Worker worker)
+        {
+            if (request.SubmissionState == null) return;
+
+            var stringToSubmissionState = new Dictionary<string, SubmissionState> {
+                { "in_progress", SubmissionState.InProgress },
+                { "submitted", SubmissionState.Submitted },
+                { "approved", SubmissionState.Approved}
+            };
+
+            if (!stringToSubmissionState.ContainsKey(request.SubmissionState.ToLower()))
+            {
+                throw new UpdateSubmissionException($"Invalid submission state supplied {request.SubmissionState}");
```

Comment:
is this ever user-supplied, or is it determined from the attributes you do pass in?

Reply:
request.SubmissionState comes from the request, if it isn't a valid submission state as determined by this if check we throw an error.